### ISSUE TITLE
Maximum update depth exceeded

### ISSUE THREAD

**Issue by thekapadia** [aeagle/react-spaces#45]
@aeagle When I run the code at https://codesandbox.io/s/misty-fast-r4qeo - it works fine in this online sandbox. 

But when I run the same code on my local machine in an app set up via `create-react-app`, it throws the below error (and the browser freezes).

```
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
in SpaceInternal (created by Fill)
```

**Comment by thekapadia** [comment]
@aeagle I found out that this specifically happens when the `size` is dynamically set to `0`. 

Try this https://codesandbox.io/s/loving-wildflower-f9wjx - run this code in a React app on your local machine.

**Comment by thekapadia** [comment]
Hi @aeagle - just passing by to see if there's any update on this issue. Thanks!

**Comment by Slapbox** [comment]
@aeagle this hasn't caused me any trouble, but I did just hit this once just now. I'm not sure what, if anything, might have been different when this occurred.

```js
react-dom.development.js:546 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
    in SpaceInternal (created by Fill)
    in Fill (created by SpacesRoot)
    in div (created by Context.Consumer)
    in InsideViewport (created by SpacesRoot)
    in div (created by SpaceInternal)
    in div (created by SpaceInternal)
    in SpaceInternal (created by Fill)
    in Fill (created by ViewPort)
    in div (created by ViewPort)
    in ViewPort (created by SpacesRoot)
    in SpacesRoot (created by ConnectFunction)
    in ConnectFunction (created by Connect(SpacesRoot))
```

**Comment by aeagle** [comment]
@thekapadia @Slapbox As noted in in other issues I've been preparing an alpha release `v0.1.20-alpha.3` with major modifications which I am hoping alleviate a lot of these issues. Feedback much welcome to see if this alleviates this issue.

**Comment by thekapadia** [comment]
@aeagle The issue of setting `size={0}` is not occurring in `v0.1.20-alpha.3`. Looks like it's fixed.

Thank you @aeagle for all your hard work and the swift releases. Really appreciate.

**Comment by aeagle** [comment]
@thekapadia Thanks for the feedback. I'll close once I get the full release out.

**Comment by aeagle** [comment]
Fix released on `0.1.20` now.
